common/http/http.cpp

```diff
@@ -324,5 +326,29 @@ void OutgoingResponse::SetBodyReader(io::ReaderPtr body_reader) {
 	body_reader_ = body_reader;
 }
 
+ExponentialBackoff::ExpectedInterval ExponentialBackoff::NextInterval() {
+	iteration_++;
+
+	if (try_count_ > 0 && iteration_ > try_count_) {
+		return expected::unexpected(MakeError(MaxRetryError, "Exponential backoff"));
+	}
+
+	chrono::milliseconds current_interval = smallest_interval_;
+	for (int count = 0; count < iteration_; count++) {
+		if (count != 0 && count % 3 == 0) {
+			auto new_interval = current_interval * 2;
+			if (new_interval > max_interval_) {
+				new_interval = max_interval_;
+			}
+			if (new_interval == current_interval) {
```

Comment:
This condition is  worth a comment, I think. It's not entirely clear at the first sight that no change means maximum reached.

Reply:
I'll just document the function better in general. I also discovered a bug while writing it: A low `try_count_` was respected, but not a high one.